### ISSUE TITLE
MarkerData::readStoFile() writes two lines to file header every time the file is read

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-core#1338]
[`MarkerData::readStoFile()`](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/MarkerData.cpp#L481) calls [`addNumRowsNumColumns()`](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Auxiliary/auxiliaryTestFunctions.h#L229), which writes
```
nRows=[number of rows]
nColumns=[number of columns]
```
to the header every time a .sto file is read. `addNumRowsNumColumns()` should write these header rows only if they don't already exist.

**Comment by klshrinidhi** [comment]
I think I added this to make sure the Tools are able to read STO file written by STOFileAdapter. I will look into this and get back.

**Comment by tkuchida** [comment]
The issue appears to start at [MarkerData.cpp#L486](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/MarkerData.cpp#L486). As we discussed, the `revertToVersionNumber1()` method can return a bool indicating whether the version number was decremented; `addNumRowsNumColumns()` should be called only if `true`:
```cpp
if (revertToVersionNumber1(aFileName, tmpFileName))
    addNumRowsNumColumns(tmpFileName, aFileName);
```

**Comment by tkuchida** [comment]
Fixed by #1349.
